src/NCShipment.php

```diff
@@ -0,0 +1,211 @@
+    {
+        $response = $this->get(
+            "rs/{$this->config['customer_number']}/ncshipment/{$shipment_id}/{$extra}",
+            ['Accept' => 'application/vnd.cpc.ncshipment-v4+xml'],
+            $options
+        );
+        return $response;
+
+    }
+
+    /**
+     * Get NCShipments from Canada Post within the specified range.
+     *
+     * @param string $from
+     *   The beginning range. YmdHs format, eg. 201808282359.
+     * @param string $to
+     *   The end range, defaults to current time. YmdHs format, eg. 201808282359.
+     * @param string $tracking_pin
+     *   The Tracking PIN of the shipment to retrieve.
+     * @param array $options
+     *   The options array.
+     *
+     * @see https://www.canadapost.ca/cpo/mc/business/productsservices/developers/services/onestepshipping/onestepshipments.jsf
+     * @return \DOMDocument|\Psr\Http\Message\StreamInterface
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     */
+    public function getNCShipments($from = '', $to = '', $tracking_pin = '', array $options = [])
+    {
+        if (empty($to)) {
+            $to = date('YmdHs');
+        }
+        $query_params = "from={$from}&to{$to}";
+
+        if (!empty($tracking_pin)) {
+            $query_params = "trackingPIN={$tracking_pin}";
+        }
+
+        $response = $this->get(
+            "rs/{$this->config['customer_number']}/ncshipment?" . $query_params,
+            ['Accept' => 'application/vnd.cpc.ncshipment-v4+xml'],
+            $options
+        );
+        return $response;
+
+    }
+
+    public function requestNCShipmentRefund($shipment_id, $email, $options) {
+        $content = [
+            'email' => $email
+        ];
+
+        $xml = Array2XML::createXML('non-contract-shipment-refund-request', $content);
+        $envelope = $xml->documentElement;
+        $envelope->setAttribute('xmlns',
+            'http://www.canadapost.ca/ws/ncshipment-v4');
+        $payload = $xml->saveXML();
+        $response = $this->post(
+            "rs/{$this->config['customer_number']}/ncshipment/{$shipment_id}/refund",
+            [
+                'Content-Type' => 'application/vnd.cpc.ncshipment-v4+xml',
+                'Accept' => 'application/vnd.cpc.ncshipment-v4+xml',
+            ],
+            $payload,
+            $options
+        );
+        return $response;
+    }
+
+    /**
+     * Helper function to build the content array.
+     *
+     * @param array $sender
+     *   The sender array.
+     * @param array $destination
+     *   The destination array.
+     * @param array $parcel
+     *   The parcel array.
+     * @param array $options
+     *   The options array.
+     *
+     * @return array
+     *   The content array.
+     */
+    protected function buildShipmentArray(
+        array $sender,
+        array $destination,
+        array $parcel,
+        array $options = []
+    ) {
+        $this->verifyPostalCode($sender);
+        $this->verifyPostalCode($destination);
```

Comment:
The `verifyPostalCode` method expects `$address['postal_code']`, `$address['address-details']['postal-zip-code']` given here.

Reply:
Also, see issue with passing by reference in the method definition.